R/spark_apply_bundle.R

```diff
@@ -67,6 +67,15 @@ spark_apply_bundle_file <- function(packages, base_path, session_id = NULL) {
 #'
 #' @export
 spark_apply_bundle <- function(packages = TRUE, base_path = getwd(), session_id = NULL) {
+  # If session_id is not provied use a random string to avoid file name collision.
+  if (is.null(session_id)) {
+    session_id <- substr(
+      digest::digest(paste(Sys.time(), collapse = "-"), algo = "sha256"),
+      start = 1,
+      stop = 7
+    )
```

Comment:
```suggestion
  session_id <- session_id %||% uuid::UUIDgenerate(use.time = TRUE)
```
I think we should just use time-based UUID here, a well-established standard for ensuring global uniqueness, which contains 60 bits worth of randomness from timestamp, and other required bits for ensuring uniqueness from clock seq no and node ID (because apparently just timestamp alone is not sufficient).

The `uuid` package is already imported by `sparklyr` btw because it's needed in a few other places in `sparklyr`.

Reply:
Actually, unless there is definitely a valid reason for not using UUID, I'll just take the liberty of committing this suggested change, wait for tests to pass, and then merge the squash of both commits.